src/lib/bounded-contexts/iam/authentication/tests/__tests__/change-email/change-email.steps.ts

```diff
@@ -0,0 +1,169 @@
+import { mockAsyncLocalStorageGet } from '../../../../../../../../test/mocks/mockAsynLocalStorageGet.mock';
+import { Application, Domain } from '@bitloops/bl-boilerplate-core';
+import {
+  CHANGE_EMAIL_SUCCESS_CASE,
+  INVALID_EMAIL_CASE,
+  USER_NOT_FOUND_CASE,
+  USER_REPO_ERROR_GETBYID_CASE,
+  USER_REPO_ERROR_UPDATE_CASE,
+} from './change-email.mock';
+import { MockUserWriteRepo } from './change-email-write-repo.mock';
+import { ChangeEmailCommand } from '../../../commands/change-email.command';
+import { ChangeEmailHandler } from '../../../application/command-handlers/change-email.handler';
+import { UserEntityBuilder } from '../../builders/user-entity.builder';
+import { UserEntity } from '../../../domain/UserEntity';
+import { DomainErrors } from '../../../domain/errors';
+import { ApplicationErrors } from '../../../application/errors';
+import { UserUpdatedEmailDomainEvent } from '../../../domain/events/user-updated-email.event';
+
+describe('Change user email feature test', () => {
+  it('Changed user email successfully', async () => {
+    const { id, email, password } = CHANGE_EMAIL_SUCCESS_CASE;
+    mockAsyncLocalStorageGet(id);
+
+    // given
+    const mockUserWriteRepo = new MockUserWriteRepo();
+    const changeEmailCommand = new ChangeEmailCommand({
+      email,
+      userId: id,
+    });
+
+    // when
+    const changeEmailHandler = new ChangeEmailHandler(
+      mockUserWriteRepo.getMockUserWriteRepo(),
+    );
+    const result = await changeEmailHandler.execute(changeEmailCommand);
+
+    //then
+    const userEntity = new UserEntityBuilder()
+      .withId(id)
+      .withEmail(email)
+      .withPassword(password)
+      .build();
+
+    expect(mockUserWriteRepo.mockGetByIdMethod).toHaveBeenCalledWith(
+      new Domain.UUIDv4(id),
+    );
+    expect(mockUserWriteRepo.mockUpdateMethod).toHaveBeenCalledWith(
+      expect.any(UserEntity),
+    );
+    const userAggregate = mockUserWriteRepo.mockUpdateMethod.mock.calls[0][0];
+    expect(userAggregate.domainEvents[0]).toBeInstanceOf(
+      UserUpdatedEmailDomainEvent,
+    );
+    expect(userAggregate.id).toEqual(userEntity.id);
+    expect(userAggregate.email).toEqual(userEntity.email);
+    expect(userAggregate.password).toEqual(userEntity.password);
```

Comment:
or maybe create a UserPropsBuilder instead and check that the aggregate.props are equal with the props of the builder.